### PR TITLE
fix: contentlanguages need to be matched before piping output to target

### PR DESCRIPTION
**Pull request by m90**
Connects https://github.com/maxlath/wikibase-cli/issues/188

The script is a bit all over the place (i.e. it's doing two totally unrelated things at once), but I felt it's ok since the surface area is still very small and future readers might only be interested in the fact that data is "somehow mangled".

### CODE REVIEW

File: adjust_data.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import argparse
+import json
+import fileinput
+
+parser = argparse.ArgumentParser(
+    description="Adjust a line from wb-cli so it can be fed into the target wiki"
+)
+
+parser.add_argument("-s", "--source", action="store", dest="source", required=True)
```
Comment (m90) [inline adjust_data.py]: Thinking about it again, we probably don't even need the target languages as we implicitly get these when requesting the entities. The for loop will filter what's not to be passed, so no need to intersect the sets.

File: mangle_data.py
Comment (tarrow) [inline mangle_data.py:1]: A most amusing name

File: mangle_data.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+parser.add_argument("-t", "--target", action="store", dest="target", required=True)
+parser.add_argument("-p", "--pick", action="append", dest="pick", required=True)
+
+
+def process_line(line, pick=[], select_languages=set()):
+    data = json.loads(line)
+    out = {}
+    for key, value in data.items():
+        if key not in pick:
```
Comment (tarrow) [inline mangle_data.py:19]: in this case `pick` can have values like `label` `description` etc. right?

File: mangle_data.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            continue
+
+        if not isinstance(value, dict):
+            out[key] = value
```
Comment (tarrow) [inline mangle_data.py:23]: so this is basically "skip if there are no <labels|descriptions>" etc?

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,9 @@
 FROM node:20-alpine
 
-RUN apk add --no-cache jq=1.7.1-r0 curl=8.8.0-r0 && \
+RUN apk add --no-cache curl=8.8.0-r0 python3=3.12.3-r1 && \
```
Comment (tarrow) [inline Dockerfile:3]: i'd not noticed this removed the jq dependency; that's a nice happy accident

File: mangle_data.py (hunk 2 above)
Comment (tarrow) [reply mangle_data.py:19]: (yes, I saw this below in where it is used)

File: mangle_data.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
```
Comment (tarrow) [inline mangle_data.py:1]: I have no strong objection to us using python here; I wonder (if we're going to keep using it); if we'd benefit from throwing a light sprinkling of all the usual tooling here: some linting, maybe even some super light e2e test ?

What do you think?

Comment (m90) [reply mangle_data.py:1]: I think I would prefer not having to introduce dependency management for Python here, but having a small test case written as a shell script might be a nice addtion, yes. I'll see what I can come up with.

File: mangle_data.py (hunk 3 above)
Comment (m90) [reply mangle_data.py:23]: Yes.

File: test.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+import unittest
```
Comment (tarrow) [inline test.py:3]: living in the future eh?

**Review by tarrow — APPROVED** [review]

**Pull request merged**